scripts/release/asset_publish.py

```diff
@@ -179,26 +178,47 @@ def validate_update_command_component(
 
     env = None
     # Check if component's env is registered
-    for version in [env_version, final_version]:
-        try:
-            if (env := mlclient.environments.get(name=env_name, version=version)) is not None:
-                break
-        except Exception as e:
-            logger.print(f"Fetching component env {env_name}:{version} from registry failed. Error: {e}")
+    registered_envs = get_registered_asset_versions(assets.AssetType.ENVIRONMENT.value, env_name, registry_name)
+    env = next((x for x in registered_envs if x['version'] in [env_version, final_version]), None)
 
     if not env:
         logger.print(f"Could not find a registered env for {component.name}. Please retry again!!!")
         return False
 
-    component.environment = env.id
+    # TODO: Bug in env list, which does not give complete detail
+    # https://github.com/Azure/azure-sdk-for-python/issues/29248
+    env_id = f"azureml://registries/{registry_name}/environments/{env['name']}/versions/{env['version']}"
+    logger.print(f"Updating component env to {env_id}")
+    component.environment = env_id
     if not update_spec(component, spec_path):
         logger.print(f"Component update failed for asset spec path: {asset.spec_path}")
         return False
+    return True
+
+
+def run_command(cmd: List[str]):
+    """Run the command for and return result."""
+    result = run(cmd, stdout=PIPE, stderr=PIPE, encoding=sys.stdout.encoding, errors="ignore")
+    return result
+
+
+def asset_list_command(
```

Comment:
This is only called once. Consider just creating the `cmd` list in `get_registered_asset_versions`.